src/Raven.Client/Exceptions/UnsuccessfulFileAccessException.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace Raven.Client.Exceptions
+{
+    public class UnsuccessfulFileAccessException : Exception
+    {
+        private static string GetMessage(string filePath) => 
+            $@"Failed to access a file at <{filePath}>. There are multiple possible reasons: {Environment.NewLine}
+* RavenDB process has insufficient permissions to access the file. Please verify OS file permissions. {Environment.NewLine}
+* The file is being used by some other process. {Environment.NewLine}
```

Comment:
I wonder if we could use `WhoIsLocking.GetProcessesUsingFile` to check that and provide a blocking process name if that's the case (not that for it always returns empty list on Posix)

Reply:
@arekpalinski That would only work on Windows (although that is the only thing that will also be locking in this manner)